uniro_admin_frontend/src/page/buildingPage.tsx

```diff
@@ -73,70 +105,153 @@ const BuildingPage = () => {
-          setSelectedBuildingId(0);
-          return;
-        });
-      }
-    },
-    [Polyline, AdvancedMarker, map]
-  );
+      //polyline을 선택 방지
+      google.maps.event.addListener(
+        polyline,
+        "click",
+        (e: { latLng: google.maps.LatLng }) => {
+          if (mode === "view" || mode === "add") setSelectedBuildingId(0);
+          if (mode === "connect") {
+            const edges: CoreRoute[] = subRoutes.map(
+              ({ routeId, node1, node2 }) => {
+                return { routeId, node1, node2 };
+              }
+            );
+
+            const { edge: nearestEdge } = findNearestSubEdge(edges, {
+              lat: e.latLng.lat(),
+              lng: e.latLng.lng(),
+            });
+            const { node1, node2 } = nearestEdge;
+
+            const polyline = new Polyline({
+              map: map,
+              path: [
+                { lat: node1.lat, lng: node1.lng },
+                { lat: node2.lat, lng: node2.lng },
+              ],
+              strokeColor: "#000000",
+              zIndex: 100,
+            });
+
+            const marker1 = new AdvancedMarker({
+              map: map,
+              position: { lat: node1.lat, lng: node1.lng },
+              content: createMarkerElement({
+                type: "waypoint_red",
+                className: "translate-waypoint",
+              }),
+              zIndex: 100,
+            });
+
+            const marker2 = new AdvancedMarker({
+              map: map,
+              position: { lat: node2.lat, lng: node2.lng },
+              content: createMarkerElement({
+                type: "waypoint_blue",
+                className: "translate-waypoint",
+              }),
+              zIndex: 100,
+            });
+            setSelectedEdge({
+              info: nearestEdge,
+              marker1,
+              marker2,
+              polyline,
+            });
+
+            setSelectedNode([node1, node2]);
+          }
+        }
+      );
+      setPolylineList((prev) => [...prev, polyline]);
 
-  const addBuildings = useCallback(() => {
+      const startNode = subRoutes[0].node1;
+
+      const startMarker = new AdvancedMarker({
+        map: map,
+        position: startNode,
+        content: createMarkerElement({
+          type: "waypoint",
+          className: "translate-waypoint",
+        }),
+      });
+
+      startMarker.addListener("click", () => {
+        if (mode === "view" || mode === "add") setSelectedBuildingId(0);
+      });
+    }
+  };
+
+  const drawSingleRoute = (start: Coord, end: Coord) => {
+    if (!Polyline || !AdvancedMarker || !map) return;
+
+    if (singleRoute) {
+      singleRoute.polyline.setMap(null);
+    }
+
+    if (selectedBuildingId === 0) {
+      return;
+    }
+
+    const polyline = new Polyline({
+      map: map,
+      path: [start, end],
+      strokeColor: "#808080",
+    });
+
+    setSingleRoute({ start, end, polyline });
```

Comment:
```suggestion
setSingleRoute(prevRoute => {
    if(prevRoute.polyline) {
        polyline.setPath([start, end]);
    }
    
    return {
        ...prevRoute
        start: start,
        end: end,
    }
})
```

- 이렇게 항상 지우고 생성하는 것이 아닌 path를 변경하여 재사용할 수 있을것 같습니다.